dag_in_context/src/optimizations/function_inlining.rs

```diff
@@ -1 +1,83 @@
+use std::{collections::HashMap, vec};
 
+use crate::schema::{Expr, RcExpr, TreeProgram};
+
+#[derive(Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct CallBody {
+    pub call: RcExpr,
+    pub body: RcExpr,
+}
+
+// Gets a list of all the calls in the program
+// and pairs them with an inlined body
+fn get_calls_and_subst(
```

Comment:
I think you should separate concerns- make a `get_calls` function and use the resulting set to do the inlining